topology/enhancers/user_metadata.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package metadata
+
+import (
+	"sync"
+
+	"github.com/skydive-project/skydive/api"
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/topology"
+	"github.com/skydive-project/skydive/topology/graph"
+)
+
+//UserMetadataManager describes user metadata manager
+type UserMetadataManager struct {
+	sync.RWMutex
+	graph.DefaultGraphListener
+	graph           *graph.Graph
+	metadataHandler *api.UserMetadataAPIHandler
+	metadata        map[string]*api.UserMetadata
+	watcher         api.StoppableWatcher
+}
+
+//OnNodeAdded event
+func (u *UserMetadataManager) OnNodeAdded(n *graph.Node) {
+	u.nodeEvent()
+}
+
+//OnNodeUpdated event
+func (u *UserMetadataManager) OnNodeUpdated(n *graph.Node) {
+	u.nodeEvent()
+}
+
+func (u *UserMetadataManager) nodeEvent() {
+	//u.RLock()
```

Comment:
i kept it purposefully to convey to you ppl that i used this to find out the data race :) will remove in the upcoming version.